src/test/java/com/softserve/bookstore/controllers/BookControllerTest.java

```diff
@@ -0,0 +1,165 @@
+package com.softserve.bookstore.controllers;
+
+import com.softserve.bookstore.models.Author;
+import com.softserve.bookstore.models.Book;
+import com.softserve.bookstore.models.Genre;
+import com.softserve.bookstore.service.BookService;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.ResultActions;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+class BookControllerTest extends BaseControllerTest<Book> {
+
+    private static final String FILE_NAME = "src/main/resources/books";
+    @MockBean
+    BookService bookService;
+
+    @Test
+    @DisplayName("Finding book")
+    void findBook_IF_Success() throws Exception {
+        Book book = new Book(1, "STrula", new Author(1, " Bula ", "Mih223ai"),
+                Genre.FICTION, 12000);
+
+        List<Book> books = new ArrayList<>();
+        books.add(book);
+
+        when(bookService.findAllBooks()).thenReturn(books);
+
+        ResultActions resultActions = mockMvc.perform(get("/books"));
+
+        resultActions.andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$", hasSize(1)))
+                .andExpect(jsonPath("$[0].idBook", is(book.getIdBook())));
+
+        assertThat(book.getIdBook()).isPositive();
+        assertThat(book.getIdBook()).isEqualTo(1);
+        assertNotNull(books);
+
+    }
+
+    @Test
+    void getAlBookFromTxT_If_Success() throws Exception {
+
+
+        List<Book> booksListFromFile = List.of(new Book(1, "White Fang", new Author("George", "Martin"), Genre.HORROR, 1200));
+        when(bookService.getBooksFromFile(FILE_NAME)).thenReturn(booksListFromFile);
+
+        ResultActions resultActions = mockMvc.perform(get("/books/file"));
+        resultActions.andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$", hasSize(1)))
+                .andExpect(jsonPath("$[0].idBook").isNotEmpty())
+                .andExpect(jsonPath("$[0].title").value("White Fang"));
+
```

Comment:
as you might see in this case using blank line is good
as you use blank line to divide logic (in this case expects and asserts)